### PR TITLE
Implement projection fields assignment

### LINKED ISSUE #2446: Support field projections assignment
### What should be done?

We should support fields assignment with projection as shown in the example below.

```js
db.values.find({}, { v: '123'})
[
  { _id: 'datetime', v: '123' },
  { _id: 'int64-max', v: '123' },
  { _id: 'double-2', v: '123' },
  { _id: 'string-whole', v: '123' },
  { _id: 'binary', v: '123' },
  { _id: 'regex', v: '123' },
  { _id: 'int32-zero', v: '123' },
  { _id: 'double-max', v: '123' },
]
```

### Where?

internal/handlers/common/projection.go

### Definition of Done

- all handlers updated;
- unit tests added/updated;
- integration/compatibility tests added/updated;
- spot refactorings done;
- user documentation updated;
- something else?

### PR DESCRIPTION
**Pull request by w84thesun**
# Description

Closes #2446.

## Readiness checklist

* [ ] I added/updated unit tests.
* [x] I added/updated integration/compatibility tests.
* [x] I added/updated comments and checked rendering.
* [ ] I made spot refactorings.
* [ ] I updated user documentation.
* [ ] I ran `task all`, and it passed.
* [x] I ensured that PR title is good enough for the changelog.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

File: integration/query_projection_compat_test.go [hunk 1]
```diff
@@ -100,6 +102,28 @@ func TestQueryProjectionCompat(t *testing.T) {
...
+		},
+		"Assign2FieldsIncludeID": {
+			filter:     bson.D{},
+			projection: bson.D{{"_id", true}, {"foo", nil}, {"bar", "qux"}},
+		},
+		"Assign1FieldExcludeID": {
+			filter:      bson.D{},
+			projection:  bson.D{{"_id", false}, {"foo", primitive.Regex{Pattern: "^fo"}}},
+			skipIDCheck: true,
+		},
```
Comment (chilagrow) [inline integration/query_projection_compat_test.go:126]: Just noticed we don't have test case for testing exclusion and inclusion both. I think below two should error but give different error message. Could you add them please?
```
"Exclusion": {
			filter:     bson.D{},
			projection: bson.D{{"foo", false}, {"bar", true}},
		},
```

```
"Inclusion": {
			filter:     bson.D{},
			projection: bson.D{{"foo", true}, {"bar", false}},
		},
```

**Review by chilagrow — COMMENTED** [review]
Just a single comment while looking at tests :hugs:

File: integration/query_projection_compat_test.go (hunk 1 above)
Comment (chilagrow) [reply integration/query_projection_compat_test.go:132]: Perfect thanks!

**Review by noisersup — APPROVED** [review]
:+1:

**Pull request merged**